### PR TITLE
Minor Dungeon Notes & Trap Function

### PR DESCRIPTION
**Pull request by Jawshoeuh**
Add GendersOppositeNotGenderless and TryRemoveTrap. Minor dungeon notes, adding power point boost flags to  moves struct, and adding hidden power to struct team_member.

### CODE REVIEW

File: symbols/overlay29.yml [hunk 1]
```diff
@@ -6014,7 +6025,9 @@ overlay29:
         NA: 0x2321104
         JP: 0x23225B0
       description: |-
-        Checks that the given monster is standing on a tile it can stand on given its movement type, and warps it to a random location if it's not.
+        Checks that the given monster is standing on a tile it can stand on given its
+        movement type. If the monster is a non-leader ally, they will be warped to the
+        leader. Otherwise, the monster is warped randomly.
```
Comment (Frostbyte0x70) [inline symbols/overlay29.yml:6030]: This new description no longer states that the warping only takes place if the monster cannot stand on the tile

File: headers/functions/arm9.h [hunk 2]
```diff
@@ -974,6 +974,7 @@ bool BaseFormsEqual(enum monster_id monster1, enum monster_id monster2);
 bool DexNumbersEqual(enum monster_id monster1, enum monster_id monster2);
 bool GendersEqual(enum monster_id monster1, enum monster_id monster2);
 bool GendersEqualNotGenderless(enum monster_id monster1, enum monster_id monster2);
+bool GendersOppositeNotGenderless(enum monster_id monster1, enum monster_id monster2);
```
Comment (UsernameFodder) [inline headers/functions/arm9.h:977]: Can we call this GendersNotEqualNotGenderless for consistency with GendersEqualNotGenderless?

File: headers/types/dungeon_mode/dungeon.h [hunk 3]
```diff
@@ -1691,13 +1691,12 @@ struct dungeon {
     undefined field_0x2caf5;
     undefined field_0x2caf6;
     undefined field_0x2caf7;
-    // 0x2CAF8: Related to playing dungeon music? Initialized to 0x4.
+    // 0x2CAF8: Related to playing dungeon music? Initialized to 0x0 at the
```
Comment (UsernameFodder) [inline headers/types/dungeon_mode/dungeon.h:1694]: This used to say initialized to 0x4, now it says 0x0. Is this a correction or a mistake?

File: headers/types/dungeon_mode/dungeon_mode_common.h [hunk 4]
```diff
@@ -53,9 +53,10 @@ struct move {
     // Consume 4 PP this turn. No effect unless f_consume_pp is set, overrides f_consume_2_pp
     bool f_consume_4_pp : 1;
     uint16_t flags2_unk7 : 1;
-    uint16_t flags2_unk8 : 1;
+    bool f_multitalent_iq_pp_boost : 1; // The IQ skill Multitalent is actively boosting move PP
```
Comment (UsernameFodder) [inline headers/types/dungeon_mode/dungeon_mode_common.h:56]: This name makes it sound like multitalent is boosting IQ. Can we just call it `f_multitalent_pp_boost`? I think "multitalent" is distinct enough on its own.

**Review by UsernameFodder — CHANGES_REQUESTED** [review]

File: headers/types/dungeon_mode/dungeon.h (hunk 3 above)
Comment (Jawshoeuh) [reply headers/types/dungeon_mode/dungeon.h:1694]: This is a correction.

**Review by UsernameFodder — APPROVED** [review]

**Pull request merged**